src/automerge.js

```diff
@@ -95,9 +96,10 @@ function getAllChanges(doc) {
   return getChanges(init(), doc)
 }
 
-function applyChanges(doc, changes) {
+function applyChanges(doc, changes, options = {}) {
   const oldState = Frontend.getBackendState(doc)
   const [newState, patch] = Backend.applyChanges(oldState, changes)
+  if (options.patchCallback) options.patchCallback(Object.assign({}, patch))
```

Comment:
I would prefer a operate line + indentation, which improves readability, safety and debuggablility.